attribute/genericattribute.go

```diff
@@ -17,48 +17,52 @@ type GenericAttribute struct {
 
 // NewGeneric creates a new generic attribute
 func NewGeneric(a *yotiprotoattr.Attribute) *GenericAttribute {
-	var value interface{}
+	value := parseValue(a.ContentType, a.Value)
+	parsedAnchors := anchor.ParseAnchors(a.Anchors)
+
+	return &GenericAttribute{
+		Attribute: &yotiprotoattr.Attribute{
+			Name:        a.Name,
+			ContentType: a.ContentType,
+		},
+		value:   value,
+		anchors: parsedAnchors,
+	}
+}
 
-	switch a.ContentType {
+func parseValue(contentType yotiprotoattr.ContentType, byteValue []byte) (result interface{}) {
+	switch contentType {
 	case yotiprotoattr.ContentType_DATE:
-		parsedTime, err := time.Parse("2006-01-02", string(a.Value))
+		parsedTime, err := time.Parse("2006-01-02", string(byteValue))
 		if err == nil {
-			value = &parsedTime
+			result = &parsedTime
 		} else {
-			log.Printf("Unable to parse date value: %q. Error: %q", string(a.Value), err)
+			log.Printf("Unable to parse date value: %q. Error: %q", string(byteValue), err)
 		}
 
 	case yotiprotoattr.ContentType_JSON:
-		unmarshalledJSON, err := UnmarshallJSON(a.Value)
+		unmarshalledJSON, err := UnmarshallJSON(byteValue)
 
 		if err == nil {
-			value = unmarshalledJSON
+			result = unmarshalledJSON
 		} else {
-			log.Printf("Unable to parse JSON value: %q. Error: %q", string(a.Value), err)
+			log.Printf("Unable to parse JSON value: %q. Error: %q", string(byteValue), err)
 		}
 
 	case yotiprotoattr.ContentType_STRING:
-		value = string(a.Value)
+		result = string(byteValue)
 
 	case yotiprotoattr.ContentType_JPEG,
 		yotiprotoattr.ContentType_PNG,
+		yotiprotoattr.ContentType_MULTI_VALUE,
 		yotiprotoattr.ContentType_UNDEFINED:
-		value = a.Value
+		result = byteValue
```

Comment:
I'm not convinced by this.
Here you need a separate case that triggers the recursive call.
If it's an image, you correctly return the binary, but if it's a MULTI_VALUE, you need to do the following:

1. Unmarshal the binary a MutliValue protobuf https://github.com/lampkicking/core-api/blob/master/attrpubapi_v1/Attribute.proto#L39
2. Create a slice that is as long as the number of `values` in the unmarshalled proto.
3. Iterate through the values and invoke `parseValue` passing the `contentType` and `data`  from the i-th value. 
4. Assign the result of the call to the corresponding element in the slice.